lib/arclight/repository.rb

```diff
@@ -59,11 +59,18 @@ def self.all
       from_yaml(ENV['REPOSITORY_FILE'] || 'config/repositories.yml').values
     end
 
-    # Mimics ActiveRecord `find_by` behavior
+    # Mimics ActiveRecord dynamic `find_by` behavior for the slug or name
     #
-    # @param [String] `slug`
-    def self.find_by(slug)
-      all.select { |repo| repo.slug == slug }.first
+    # @param [String] `slug` or `name`
+    # @return [Repository]
+    def self.find_by(slug: nil, name: nil)
+      if slug
+        all.select { |repo| repo.slug == slug }.first
```

Comment:
`detect` could also be used (as they are the same method).

https://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find